### PR TITLE
feat(consul): Adds a data source to retrieve kv pairs under a specified path

### PR DESCRIPTION
**Pull request by samber**
Hi guys,

Example: 

```
provider "consul" {
    address = "1.2.3.4:8500"
    datacenter = "eu_central_1"
}

data "consul_key_prefix" "test" {
     path_prefix = "services/api/"
}

output "variables" { value = "${data.consul_key_prefix.test.var}" }
```

Read have been deprecated in resource consul_keys after adding the consul_keys data source. I think we should do the same for the resource consul_key_prefix. This data source should be the correct way to get data.

### CODE REVIEW

**Comment by stack72** [comment]
Hi @samber 

Thanks for the work here - looks like this doesn't build - any chance you could have a look at it?

Thanks

Paul

**Comment by samber** [comment]
@stack72 sorry for the late fix.

This is working well ;)

Should we deprecate reads from consul_key_prefix resource ?

**Comment by partsalliance** [comment]
@stack72 any ideas when this PR can be merged?

**Comment by samber** [comment]
up ?

**Comment by nanoz** [comment]
Having this PR checked and merged would be very helpful to us @stack72 :)

File: builtin/providers/consul/data_source_consul_key_prefix.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	if err := d.Set("var", vars); err != nil {
+		return err
+	}
+
+	// Store the datacenter on this resource, which can be helpful for reference
+	// in case it was read from the provider
+	d.Set("datacenter", dc)
+
+	d.SetId("-")
```
Comment (stack72) [inline builtin/providers/consul/data_source_consul_key_prefix.go]: If there is no Id, then we should use something like:

```
d.SetId(time.Now().UTC().String())
```

**Comment by apparentlymart** [comment]
Note that `consul_key_prefix` has never supported read-only usage, so there's no need to deprecate anything there as we did with `consul_keys`.

**Comment by teamterraform** [comment]
Hi folks! Now that terraform 0.12 is released, we are undertaking a general GH issue/PR cleanup. The providers now live in their own repositories, so any new issues or PRs should be opened there. Thanks!

**Pull request closed without merging**